### PR TITLE
add docs

### PR DESCRIPTION
**Pull request by jkhelil**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Jenkins Operator
+
+The easiest way to deploy, manage and upgrade jenkins on kubernetes.
```
Comment (akram) [inline docs/README.md]: Jenkins

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+This operator is typically installed from the Operator Lifecycle Manager (OLM), and defines jenkins CustomResourcesDefinition (CRDs) that act as
+building block for jenkins intance.
+
+See the following topics to learn how to use the jenkins operator:
+
+- [Insalling the operator](install.md)
```
Comment (akram) [inline docs/README.md]: Installing

File: docs/configure.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# Configure Jenkins in the Jenkins CR
+Wen deploying jenkins via the opertor, you can enable configuration as code for jenkins.
```
Comment (akram) [inline docs/configure.md]: s/jenkins/Jenkins/g

s/opertor/operator/

File: docs/configure.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+When enabled, the operator will add volumeMounts from the configmap configurations you provide, to the an init container inside the jenkins pod.
+The role of the init container is to copy the configuration as code files you provide into a target configuration as code volume, mounted to the jenkins container. This way we have a read write volume on the jenkins container that holds your configuration as code files.
+Another Sidecar container is also created if you choose to enable autoreload of the configuration as code files. That way, if you change the configmaps for jenkins configuration, The jenkins configuration is reloaded.
+
```
Comment (akram) [inline docs/configure.md]: This should go in a specific section like: "How it works?"

File: docs/configure.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+The jenkins CR exposes attributes to configure jenkins, you can specify the following in the jenkins CR spec:
+* enabled: speicifies if the configuration as code will be enabled
```
Comment (akram) [inline docs/configure.md]: use "-" to create a list
and backquotes for formatting

File: docs/configure.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+* defaultConfig: specifies if you want the default config for jenkins. See the the default configmap under deploy/default-config.yaml
+* configurations: an array of configuration profiles(configmap names used to configure jenkins)
+* enableAutoReload: specifies if you want to enable configuration auto reload.
+
+Here is an some example of jenkins configmaps:
```
Comment (akram) [inline docs/configure.md]: Here is an example of jenkins `configMaps`:

File: docs/install.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# Installing Jenkins Operator
+
+This guide explains how to install jenkins operaotr.
```
Comment (akram) [inline docs/install.md]: s/operaotr/operator/g

File: docs/README.md (hunk 1 above)
Comment (waveywaves) [reply docs/README.md]: also Kubernetes as these are nouns

File: docs/deploy.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+# Deploying Jenkins
+
+This guide explains how to deploy jenkins once Jenkins Operator is up and running.
+
+## Create The Jenkins CR
+
+Create a new jenkins CR or use an existing example under deploy/crds/jenkins_v1alpha2_jenkins_cr.yaml.
+For openshift you can use deploy/crds/openshift_jenkins_v1alpha2_jenkins_cr.yaml.
+
+## Deploy a Jenkins to Kubernetes
```
Comment (waveywaves) [inline docs/deploy.md:10]: *Deploy a Jenkins instance to Kubernetes

File: docs/deploy.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    ```bash
+    kubectl apply -f https://raw.githubusercontent.com/jenkinsci/kubernetes-operator/master/deploy/crds/jenkins_v1alpha2_jenkins_cr.yaml
+    ```
+
+## Deploy a Jenkins to openshift
```
Comment (waveywaves) [inline docs/deploy.md:15]: *Deploy a Jenkins instance to Openshift

File: docs/install.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+
+## Requirements
+
+To run Jenkins Operator, you will need:
+  * access to a Kubernetes cluster version 1.11+
+  * kubectl version 1.11+
+
+## Configure Jenkins Custom Resource Definition
```
Comment (waveywaves) [inline docs/install.md]: This can be called "Prerequisites" instead as installing the CRD is a prereq to installing  the operator

**Review by waveywaves — CHANGES_REQUESTED** [review]

**Pull request merged**